pkg/dataobj/internal/dataset/reader_basic.go

```diff
@@ -226,7 +226,7 @@ func (pr *basicReader) fill(ctx context.Context, columns []Column, s []Row) (n i
 			for i := columnRead; i < int64(maxRead); i++ {
 				// Reset values to 0 without discarding any memory they are pointing to
 				if !s[n+int(i)].Values[columnIndex].IsNil() {
-					s[n+int(i)].Values[columnIndex].num = 0
+					s[n+int(i)].Values[columnIndex].Zero()
```

Comment:
For the future:

Right now, this will cause things to be inconsistently reported as NULL or the zero value for the column (depending on what happens to be in the Value at the time Zero is called). I think that's fine for the moment since we don't distinguish between them, but it could be the source of the bug in the future.

Could we add a comment here explaining this behaviour?